aws/resource_aws_media_connect_flow_test.go

```diff
@@ -0,0 +1,290 @@
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSMediaConnectFlowConfig_Options(t *testing.T) {
+	var flow mediaconnect.Flow
+
+	rName := fmt.Sprintf("tfacctest%s", acctest.RandString(5))
+	resourceName := "aws_media_connect_flow.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMediaConnectFlowDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSMediaConnectFlowConfig_Options(rName, "us-west-2a", "Test Description1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMediaConnectFlowExists(resourceName, &flow),
+					resource.TestCheckResourceAttr(resourceName, "availability_zone", "us-west-2a"),
+					resource.TestCheckResourceAttr(resourceName, "source.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "source.0.description", "Test Description1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSMediaConnectFlowExists(resourceName string, flow *mediaconnect.Flow) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No  Media Connect Flow ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).mediaconnectconn
+		output, err := conn.DescribeFlow(&mediaconnect.DescribeFlowInput{
+			FlowArn: aws.String(rs.Primary.ID),
+		})
+		if err != nil {
+			return err
+		}
+
+		if output == nil || output.Flow == nil {
+			return fmt.Errorf("Media Connect Flow (%s) not found", rs.Primary.ID)
+		}
+
+		if aws.StringValue(output.Flow.FlowArn) != rs.Primary.ID {
+			return fmt.Errorf("Media Connect Flow (%s) not found", rs.Primary.ID)
+		}
+
+		*flow = *output.Flow
+
+		return nil
+	}
+}
+
+func testAccCheckAWSMediaConnectFlowDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_media_connect_flow" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).mediaconnectconn
+
+		// Handle eventual consistency
+		err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+			output, err := conn.DescribeFlow(&mediaconnect.DescribeFlowInput{
+				FlowArn: aws.String(rs.Primary.ID),
+			})
+
+			if err != nil {
+				if isAWSErr(err, mediaconnect.ErrCodeNotFoundException, "") {
+					return nil
+				}
+				return resource.NonRetryableError(err)
+			}
+
+			if output != nil && output.Flow != nil && aws.StringValue(output.Flow.FlowArn) == rs.Primary.ID {
+				return resource.RetryableError(fmt.Errorf("Media Connect Flow %s still exists", rs.Primary.ID))
+			}
+
+			return nil
+		})
+
+		return err
```

Comment:
Nit: To allow the for loop to continue, this should guard against early return:

```suggestion
		if err != nil {
			return err
		}
```

Reply:
fixed 30add7e